docs-infra: update AIO dependencies

petebacondarwin opened the pull request.

petebacondarwin committed:
build(docs-infra): refresh yarn.lock

This file had not been updated for some time and lots of
the dependencies have new versions.

This is actually necessary because (at least on OS/X) there
is a problem with `chokidar` and `fsevent` that is solved by
bumping the versions here.

petebacondarwin committed:
build(docs-infra): update to latest dgeni-packages

After the previous update to yarn.lock a problem surfaced
with the `mkdir-promise` package. The latest version of
dgeni-packages (0.28.3) fixes this problem.

petebacondarwin committed:
build(docs-infra): update AIO to Angular 9.0.0

petebacondarwin commented:
Interesting that the size checks failed:

```
Verifying sizes against '/home/circleci/ng/aio/scripts/_payload-limits.json'...
SUCCESS: Commit 0682487602fcf0ef82259df8d7abfd632be07273 uncompressed runtime-es2015 did NOT cross size threshold of 500 bytes or >1% (expected: 2987, actual: 2987).
FAIL: Commit 0682487602fcf0ef82259df8d7abfd632be07273 uncompressed main-es2015 fell below expected size by 500 bytes or >1% (expected: 455897, actual: 449223).
FAIL: Commit 0682487602fcf0ef82259df8d7abfd632be07273 uncompressed polyfills-es2015 exceeded expected size by 500 bytes or >1% (expected: 52195, actual: 55321).
```

`main-es2015` went down by too much; while `polyfills-es2015` went up!

I think this is reasonable. Updating the PR.

petebacondarwin committed:
ci(docs-infra): update payload limits

The update to Angular 9.0.0 appears to have lowered the main.js
file slightly, while the current master build of Angular appears
to have gone up slightly.

mary-poppins commented:
You can preview 0682487 at https://pr35379-0682487.ngbuilds.io/.

gkalpak commented on aio/package.json:
```diff
@@ -87,28 +87,28 @@
+    "@angular/core": "9.0.0",
+    "@angular/elements": "9.0.0",
+    "@angular/forms": "9.0.0",
     "@angular/material": "9.0.0-rc.8",
```
Maybe update CDK/Material too?

petebacondarwin replied:
Let me try...

petebacondarwin replied:
Added

gkalpak commented on aio/scripts/_payload-limits.json:
```diff
@@ -3,26 +3,26 @@
-        "main-es2015": 455897,
-        "polyfills-es2015": 52195
+        "main-es2015": 449223,
+        "polyfills-es2015": 55321
```
I just noticed this: Do we know what is causing it?

petebacondarwin replied:
The old and the new files seem really quite different. The `git diff` does not help much.

petebacondarwin replied:
The reason was the update to `@webcomponents/custom-element` which I have reverted.

mary-poppins commented:
You can preview 50c9f75 at https://pr35379-50c9f75.ngbuilds.io/.

petebacondarwin committed:
build(docs-infra): update to latest Angular Material

mary-poppins commented:
You can preview f16c91f at https://pr35379-f16c91f.ngbuilds.io/.

petebacondarwin committed:
build(docs-infra): pin @webcomponents/custom-elements to 1.2.1

The previous range (^1.2.0) allowed the version 1.3.2 to be
installed which caused the ES2015 polyfills.js file to increase
in size unwantedly.

mary-poppins commented:
You can preview a26c0a0 at https://pr35379-a26c0a0.ngbuilds.io/.

kara commented on aio/scripts/_payload-limits.json:
```diff
@@ -12,7 +12,7 @@
       "uncompressed": {
         "runtime-es2015": 2987,
-        "main-es2015": 448928,
+        "main-es2015": 449438,
```
Do we know where the 500 bytes are coming from? Did we add more

petebacondarwin replied:
So even on master right now it is 449289, so this PR's change is actually really only 149 bytes. I am not sure what the difference is though... I'll look into it a bit.

petebacondarwin replied:
I looked through the diff and could not see any obvious changes that I could identify as being the culprit. I suspect it is a combination of the styling refactor and changes to Material/CDK.
